### DISCUSSION TITLE
Tenable Nessus : Authentication to Rest API of Tenable Nessus

### DISCUSSION THREAD

**Discussion by tridariel** [grafana/grafana-infinity-datasource#107] [Specific APIs]
Hello,

I would like to use this plugin in order to connect to Tenable Nessus Rest API, but i seem to be struggling with authentication. 

I need to use the REST API keys passed in header but I dont know how to konfigure the header correctly. According to the documentation the header should be in this following format:

`x-apikey: accesskey=ACCESS_KEY; secretkey=SECRET_KEY;`

ACCESS_KEY and SECRET_KEY are generated from within Nessus. However in the option of the datasource I see only this:

![image](https://user-images.githubusercontent.com/87754670/126460794-574dd739-23da-45b6-b7b5-40be542a4e80.png)

Is there a way to correctly authenticate to this API using this plugin, or is this currently a dead end?

Thanks in advance.

**Comment by Amberrion** [comment]
Nessus API uses keys in request headers (X-ApiKeys).
In the "Authentication" section, select "No Authentication" (since we will be using custom headers).
Go to the "Custom Headers" section:
Add a header:
Header: X-ApiKeys
Value: accessKey=<your_access_key>;secretKey=<your_secret_key> Replace <your_access_key> and <your_secret_key> with the actual values.
